### PR TITLE
Nil checks in the testWriter to prevent using a bad testing.TB

### PR DESCRIPTION
**Pull request by mkeeler**
Also needed to update some funcs that were taking a *testing.T to use a testing.TB. This prevents passing a nil pointer as a non-nil interface value
and thus making it impossible to detect nil before using the interfaces functions.

### CODE REVIEW

**Comment by rboyer** [comment]
What is the use case for using a `nil` `testing.TB` value with these methods?

**Comment by mkeeler** [comment]
@rboyer Anything that uses `testutil` with the `NewTestServer` function is currently segfaulting (which includes some of our enterprise snapshot agent tests) but potentially anything that is pulling in our sdk module.

If it weren't a module that we allow others to pull in I would have just removed the methods that pass through a nil pointer but as its "public" I figured it would be better to fix the crashes.

Moving to a testing.TB in those functions prevents passing a nil *testing.T as as non-nil interface value to the TestWriter creation.\

In short these are necessary to prevent anything that calls `testutil.NewTestServer` from crashing.
